dhash/module.py

```diff
@@ -383,9 +464,55 @@ async def _save_hashes(self, message: nextcord.Message):
             )
             yield h
 
+    async def _get_url_hashes(self, message: nextcord.Message):
+        for url in re.findall(URL_REGEX, message.content):
+            if not re.search(DISCORD_REGEX, url) and (
+                not self.allowed_urls or not re.search(self.allowed_urls, url)
+            ):
+                continue
+
+            try:
+                async with aiohttp.ClientSession(
+                    headers=HTTP_HEADERS,
+                    raise_for_status=False,
+                    timeout=aiohttp.ClientTimeout(30),
+                    auto_decompress=False,
+                    read_bufsize=256,
+                ) as session:
+                    async with session.get(url) as resp:
+                        if resp.status != 200:
+                            continue
+                        size = resp.headers.get("content-length")
+                        if not size or int(size) > MAX_ATTACHMENT_SIZE * 1024:
+                            continue
+
+                        type = resp.headers.get("content-type").split("/")
+                        if (
+                            len(type) != 2
+                            or type[0] != "image"
+                            or type[1] not in ALLOWED_FORMATS
+                        ):
+                            continue
+
+                        image = Image.open(BytesIO(await resp.read()))
```

Comment:
So what happens if someone manages to spoof the content-length header?

Because we do have `allowed_urls`, it should be OK, I think?